### PR TITLE
GSdx-d3d: Port/change shader code to match opengl

### PR DESCRIPTION
**Pull request by lightningterror**
Details in the commit titles/description.

I'm not sure if the second commit is needed, if not then I could drop it.

### CODE REVIEW

File: plugins/GSdx/res/tfx.fx [hunk 1]
```diff
@@ -607,11 +610,15 @@ VS_OUTPUT vs_main(VS_INPUT input)
...
+		input.z = input.z & uint(0xffffff); 
 	}
 	else if(VS_BPPZ == 2) // 16
 	{
-		input.z = input.z & 0xffff;
+		input.z = input.z & uint(0xffff);
+	}
+	else
+	{
+		input.z;
```
Comment (turtleli) [inline plugins/GSdx/res/tfx.fx]: This does nothing.

File: plugins/GSdx/res/tfx.fx [hunk 2]
```diff
@@ -607,11 +610,15 @@ VS_OUTPUT vs_main(VS_INPUT input)
 {
 	if(VS_BPPZ == 1) // 24
 	{
-		input.z = input.z & 0xffffff; 
+		input.z = input.z & uint(0xffffff); 
 	}
 	else if(VS_BPPZ == 2) // 16
 	{
-		input.z = input.z & 0xffff;
+		input.z = input.z & uint(0xffff);
```
Comment (turtleli) [inline plugins/GSdx/res/tfx.fx]: HLSL does implicit type casting.

File: plugins/GSdx/res/tfx.fx [hunk 3]
```diff
@@ -381,7 +381,10 @@ float4x4 sample_4p(float4 u)
 
 float4 sample(float2 st, float q)
 {
-	if(!PS_FST) st /= q;
+	if(PS_FST == 0)
```
Comment (turtleli) [inline plugins/GSdx/res/tfx.fx]: Expression before and after are the same.

Comment (lightningterror) [reply plugins/GSdx/res/tfx.fx]: I wanted it to match gl code, guess it's a bit of an useless change.

File: plugins/GSdx/res/tfx.fx (hunk 1 above)
Comment (lightningterror) [reply plugins/GSdx/res/tfx.fx]: What about gl code, isn't it the same as well ?

Comment (turtleli) [reply plugins/GSdx/res/tfx.fx]: No. The commit you referenced fixed an uninitialised variable issue.

File: plugins/GSdx/res/tfx.fx (hunk 3 above)
Comment (turtleli) [reply plugins/GSdx/res/tfx.fx]: AFAICT this PR doesn't actually change anything.

File: plugins/GSdx/res/tfx.fx (hunk 1 above)
Comment (lightningterror) [reply plugins/GSdx/res/tfx.fx]: Ah ok, my main reason for opening was this change.

**Pull request closed without merging**